device-registry/src/main/scala/org/genivi/sota/device_registry/GroupInfoRepository.scala

```diff
@@ -4,58 +4,75 @@
  */
 package org.genivi.sota.device_registry
 
-
-import eu.timepit.refined.api.{Refined, Validate}
 import io.circe.Json
-import org.genivi.sota.data.{GroupInfo, Namespace}
+import org.genivi.sota.data.{GroupInfo, Namespace, Uuid}
 import org.genivi.sota.device_registry.common.{Errors, SlickJsonHelper}
 import slick.driver.MySQLDriver.api._
 import org.genivi.sota.db.SlickExtensions._
 import org.genivi.sota.refined.SlickRefined._
-
+import GroupInfo._
+import org.genivi.sota.db.Operators._
 import scala.concurrent.ExecutionContext
 
 object GroupInfoRepository extends SlickJsonHelper {
-  import GroupInfo._
 
   // scalastyle:off
   class GroupInfoTable(tag: Tag) extends Table[GroupInfo] (tag, "DeviceGroup") {
-    def groupName = column[Name]("group_name", O.PrimaryKey)
-    def namespace = column[Namespace]("namespace", O.PrimaryKey)
+    def id        = column[Uuid]("id", O.PrimaryKey)
+    def groupName = column[Name]("group_name")
+    def namespace = column[Namespace]("namespace")
     def groupInfo = column[Json]("group_info")
 
-    def * = (groupName, namespace, groupInfo).shaped <>
+    def * = (id, groupName, namespace, groupInfo).shaped <>
       ((GroupInfo.apply _).tupled, GroupInfo.unapply)
   }
   // scalastyle:on
 
   val groupInfos = TableQuery[GroupInfoTable]
 
-
   def list(namespace: Namespace)(implicit ec: ExecutionContext): DBIO[Seq[GroupInfo]] =
-    groupInfos.filter(g => g.namespace === namespace).result
+    groupInfos
+      .filter(g => g.namespace === namespace)
+      .result
 
-  protected def getGroup(groupName: Name, namespace: Namespace) =
+  protected def getGroupByName(groupName: Name, namespace: Namespace) =
```

Comment:
we are already inside a group class, we could call these `findByName` and `findById` etc